### DISCUSSION TITLE
Trying to using routing with react app on a custom domain

### DISCUSSION THREAD

**Discussion by inkslinger5198** [community/community#83179] [New to GitHub]
Hi i have a react app that i have deployed on a custom domain and it worked all fine, until I tried to route it to another page using either the browserrouter or hashrouter. I don't understand how i can route it. the page i want to route to is to be a standalone page. can someone please help me, i am attaching the following files:

### 1. Index.js
import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";

import App from "./App";
import Confirm from "./Confirm";
import "./index.css";

ReactDOM.render(

<Route path="/" element={} />
<Route path="/confirm" element={} />

,
document.getElementById("root")
);

2. app.js
import React from "react";
import { Link } from "react-router-dom";
import "./App.css";

import { Book_call, Navbar } from "./components";
import { Header, About, Service, Footer, FAQ, Contact, Founders, Testimonial } from "./containers";

const App = () => {
return (

<Book_call />

<Book_call />

);
};
export default App;

### 3. confirm.js (the page i wanted to be routed to)
import React from "react";
import "./Confirm.css";

import { Navbar_, Confirmed } from "./components";
import { Footer } from "./containers";
const Confirm = () => {
return (

    <Navbar_ />
    <Confirmed />
    <Footer />
  </div>
</div>
);
};

export default Confirm;

4. package.json
{
"homepage": "https://funnelsyndicate.co/",
"name": "funnel_syndicate2.0",
"version": "0.1.0",
"private": false,
"dependencies": {
"@calcom/embed-react": "^1.3.0",
"@testing-library/jest-dom": "^5.17.0",
"@testing-library/react": "^13.4.0",
"@testing-library/user-event": "^13.5.0",
"gh-pages": "^6.0.0",
"react": "^18.2.0",
"react-dom": "^18.2.0",
"react-icons": "^4.11.0",
"react-router-dom": "^6.21.1",
"react-scripts": "5.0.1",
"web-vitals": "^2.1.4"
},
"scripts": {
"predeploy": "npm run build",
"deploy": "gh-pages -d build",
"start": "react-scripts start",
"build": "react-scripts build",
"test": "react-scripts test",
"eject": "react-scripts eject"
},
"eslintConfig": {
"extends": [
"react-app",
"react-app/jest"
]
},
"browserslist": {
"production": [
">0.2%",
"not dead",
"not op_mini all"
],
"development": [
"last 1 chrome version",
"last 1 firefox version",
"last 1 safari version"
]
}
}

I am hosting it using gh-pages the /root directory

**Comment by rajiv2004-cloud** [comment]
Nevertheless, I'll try to help you with routing in a React app using react-router-dom.

1.index.js
import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";

import App from "./App";
import Confirm from "./Confirm";
import "./index.css";

ReactDOM.render(
  <Router>
    <Routes>
      <Route path="/" element={<App />} />
      <Route path="/confirm" element={<Confirm />} />
    </Routes>
  </Router>,
  document.getElementById("root")
);
2.app.js
import React from "react";
import { Link } from "react-router-dom";
import "./App.css";
import { Book_call, Navbar } from "./components";
import { Header, About, Service, Footer, FAQ, Contact, Founders, Testimonial } from "./containers";

const App = () => {
  return (
    <>
      <Book_call />
      <Book_call />
      {/* Your other components */}
      <Link to="/confirm">Go to Confirm Page</Link>
    </>
  );
};

export default App;
3.confrim.js
import React from "react";
import "./Confirm.css";
import { Navbar_, Confirmed } from "./components";
import { Footer } from "./containers";

const Confirm = () => {
  return (
    <>
      <Navbar_ />
      <Confirmed />
      <Footer />
    </>
  );
};

export default Confirm;


it may help you !

**Comment by rajiv2004-cloud** [comment]
Make sure that you have the correct imports in your components. Also, note that <Link to="/confirm">Go to Confirm Page</Link> is an example link to navigate to the Confirm page. Adjust the links according to your app's structure.

After making these changes, your app should be able to navigate between the "/" and "/confirm" routes. Ensure that your gh-pages deployment reflects these changes as well.
